### PR TITLE
Async IO layer + SCRAM auth

### PR DESCRIPTION
**Pull request by Nikolo**
Add async IO layer for use this module from async based app
Support kafka sasl auth (PLAIN/SCRAM) 
Move build to Dist::Zilla (milla)
Remove any test but they runs automatically from Dist::Zilla

### CODE REVIEW

File: META.json [hunk 1]
```diff
@@ -86,69 +96,115 @@
...
          "file" : "lib/Kafka/Protocol.pm",
-         "version" : "1.07"
+         "version" : "v1.700.9"
       }
    },
    "release_status" : "stable",
    "resources" : {
       "bugtracker" : {
-         "web" : "https://github.com/TrackingSoft/Kafka/issues"
+         "web" : "https://github.com/my-mail-ru/Kafka/issues"
```
Comment (asolovey) [inline META.json:174]: I think it is preferable to use main repository issue tracker - could you update your PR accordingly?

File: META.json [hunk 2]
```diff
@@ -86,69 +96,115 @@
...
-   "version" : "1.07",
-   "x_serialization_backend" : "JSON::PP version 2.27400"
+   "version" : "v1.700.9",
+   "x_contributors" : [
+      "Aleksandr Matveev <a.matveev@corp.mail.ru>",
+      "Aleksandr <timon@timon.net.nz>",
+      "Alexander Kazakov <akazakov@corp.mail.ru>",
+      "Alex Solovey <alex@trackingsoft.com>",
+      "Alex Solovey <a.solovey@gmail.com>",
+      "Alex Solovey <asolovey@users.noreply.github.com>",
```
Comment (asolovey) [inline META.json:190]: These are duplicates, and also some private addresses seem to be leaked; I understand that they are in git commit history anyway, but let's remove emails and keep unique names only.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,1036 @@
...
+Kafka package is hosted on GitHub:
+[https://github.com/TrackingSoft/Kafka](https://github.com/TrackingSoft/Kafka)
+
+# AUTHOR
+
+Sergey Gladkov
+
+Please use GitHub project link above to report problems or contact authors.
+
+# CONTRIBUTORS
```
Comment (asolovey) [inline README.md:1012]: You'd probably want to add yourself to this list.

File: lib/Kafka/IO/Async.pm [hunk 4]
```diff
@@ -0,0 +1,656 @@
+package Kafka::IO::Async;
+
+=head1 NAME
+
+Kafka::IO::Async - Pseudo async interface to nonblocking network communication with the Apache Kafka server with Coro.
+This module implemets the same interface that usual Kafka::IO module
```
Comment (asolovey) [inline lib/Kafka/IO/Async.pm:6]: typo: implements

File: lib/Kafka/IO/Async.pm [hunk 5]
```diff
@@ -0,0 +1,656 @@
...
+    # Closes and cleans up
+    $io->close;
+    undef $io;
+
+=head1 DESCRIPTION
+
+This module is private and should not be used directly.
+
+In order to achieve better performance, methods of this module do not
+perform arguments validation.
```
Comment (asolovey) [inline lib/Kafka/IO/Async.pm:96]: If I am not mistaken, they actually do validate arguments, just may be not too extensively ;)

**Review by asolovey — CHANGES_REQUESTED** [review]

File: lib/Kafka/IO/Async.pm (hunk 5 above)
Comment (Nikolo) [reply lib/Kafka/IO/Async.pm:96]: This is copy-paste from lib/Kafka/IO.pm

**Review by asolovey — APPROVED** [review]

**Pull request merged**